src/fastoad/openmdao/tests/test_variables.py

```diff
@@ -127,6 +128,176 @@ def test_variables(with_dummy_plugin_2):
+    rich_var = Variable(
+        "var1",
+        val=10.0,
+        units="m",
+        desc="Rich description",
+        upper=200.0,
+        lower=0.0,
+        tags={"tag1", "tag2"},
+    )
+    rich_var_list = VariableList([rich_var])
+
+    # Save original metadata keys for later comparison
+    original_metadata_keys = set(rich_var.metadata.keys())
+
+    # Create a variable with minimal metadata (init_metadata=False)
+    minimal_var = Variable("var1", val=15.0, init_metadata=False)
+    minimal_var_list = VariableList([minimal_var])
+
+    # Case 1: Update with merge_metadata=False (default)
+    test_var_list_no_fill = deepcopy(rich_var_list)
+    test_var_list_no_fill.update(minimal_var_list)
+
+    # Check that values were updated
+    assert test_var_list_no_fill["var1"].value == 15.0
+    # Description should be preserved since minimal_var has no description
+    assert test_var_list_no_fill["var1"].description == "Rich description"
+
+    # With merge_metadata=False, metadata like units, lower should be lost
+    updated_metadata_keys = set(test_var_list_no_fill["var1"].metadata.keys())
+    assert "units" not in updated_metadata_keys
+    assert "lower" not in updated_metadata_keys
+
+    # Case 2: Update with merge_metadata=True
+    test_var_list_with_fill = deepcopy(rich_var_list)
+    test_var_list_with_fill.update(minimal_var_list, merge_metadata=True)
+
+    # Check that values were updated
+    assert test_var_list_with_fill["var1"].value == 15.0
+    assert test_var_list_with_fill["var1"].description == "Rich description"
+
+    # With merge_metadata=True, all metadata should be preserved
+    filled_metadata_keys = set(test_var_list_with_fill["var1"].metadata.keys())
+
+    # All original keys should still be present
+    for key in original_metadata_keys:
+        assert key in filled_metadata_keys
+
+    # Check specific metadata values are preserved
+    assert test_var_list_with_fill["var1"].metadata["upper"] == 200.0
+    assert test_var_list_with_fill["var1"].metadata["lower"] == 0.0
+    assert "tag1" in test_var_list_with_fill["var1"].metadata["tags"]
+    assert "tag2" in test_var_list_with_fill["var1"].metadata["tags"]
+
+    # Test a variable list with multiple variables
+    var1 = Variable("var1", val=10.0, units="m", desc="Description for var1")
+    var2 = Variable("var2", val=20.0, units="kg", desc="Description for var2")
+    var_list = VariableList([var1, var2])
+
+    no_meta_var1 = Variable("var1", val=15.0, units="ft", init_metadata=False)
+    no_meta_var2 = Variable("var2", val=25.0, desc="New description", init_metadata=False)
+    no_meta_var_list = VariableList([no_meta_var1, no_meta_var2])
+
+    # Update with merge_metadata=True
+    var_list.update(no_meta_var_list, merge_metadata=True)
+
+    # Check results
+    assert var_list["var1"].value == 15.0
+    assert var_list["var1"].units == "ft"
+    assert var_list["var1"].description == "Description for var1"  # Preserved
```

Comment:
Shouldn't you test on another metadata as the standard behaviour is to preserve the description, unless merge_metadata is called?